src/components/DistanceRequest.js

```diff
@@ -86,16 +85,15 @@ const defaultProps = {
 };
 
 function DistanceRequest({transactionID, report, transaction, mapboxAccessToken, route, isEditingRequest, onSubmit}) {
-    const [shouldShowGradient, setShouldShowGradient] = useState(false);
-    const [scrollContainerHeight, setScrollContainerHeight] = useState(0);
-    const [scrollContentHeight, setScrollContentHeight] = useState(0);
     const {isOffline} = useNetwork();
     const {translate} = useLocalize();
 
+    const [optimisticWaypoints, setOptimisticWaypoints] = useState(null);
```

Comment:
so we cannot wait until the waypoints in the transaction is updated?
is the update for the transaction stored in Onyx asynchronous?

Reply:
> so we cannot wait until the waypoints in the transaction is updated?

We could but it caused flickering [issue](https://github.com/Expensify/App/pull/26307#issuecomment-1735667273)

> is the update for the transaction stored in Onyx asynchronous?

All Onyx updates/merge/set are asynchronous, the approach was discussed briefly [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1695884704479239)

Reply:
ah thank you for reminding me of the previous conversation. I completely forgot about it 🙇

Reply:
Yes now I understand why we need this separate state `optimisticWaypoints` to avoid the flickering.